crates/uv-workspace/src/settings.rs

```diff
@@ -0,0 +1,84 @@
+use serde::Deserialize;
+use std::path::PathBuf;
+
+use distribution_types::{FlatIndexLocation, IndexUrl};
+use install_wheel_rs::linker::LinkMode;
+use uv_auth::KeyringProvider;
+use uv_configuration::{ConfigSettings, IndexStrategy, PackageNameSpecifier};
+use uv_normalize::PackageName;
+use uv_resolver::{AnnotationStyle, ExcludeNewer, PreReleaseMode, ResolutionMode};
+use uv_toolchain::PythonVersion;
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct PyProjectToml {
+    pub(crate) tool: Option<Tools>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+pub(crate) struct Tools {
+    pub(crate) uv: Option<Options>,
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, Default, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Options {
+    pub quiet: Option<bool>,
+    pub verbose: Option<bool>,
+    pub native_tls: Option<bool>,
+    pub no_cache: bool,
+    pub cache_dir: Option<PathBuf>,
+    pub pip: Option<PipOptions>,
+    pub resolver: Option<ResolverOptions>,
+    pub installer: Option<InstallerOptions>,
```

Comment:
These should arguably be under `pip`. I'm not totally sure though to be honest.

Reply:
Do we currently have the same `ResolverOptions` and `InstallerOptions` at the top-level _and_ in `pip`?

Reply:
No, they're _only_ present at the top-level. Arguably, everything should be put under `pip`. The downside is that it creates a lot of nesting, and we may have options here that _are_ applicable to the non-`pip` interface.

Reply:
Hm but I see

```
pub struct PipOptions {
    ...
    pub resolver: Option<ResolverOptions>,
    pub installer: Option<InstallerOptions>,
```

Anyway... I'd rather put them in `pip` and pull them into the top-level as we know we need them? I think there are only a couple that would make sense.

Reply:
Maybe I messed that up? Will put them in `pip` and keep them there for now.